src/Microsoft.TemplateEngine.TemplateLocalizer.Core/Exceptions/JsonMemberMissingException.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.TemplateEngine.TemplateLocalizer.Core.Exceptions
+{
+    /// <summary>
+    /// Represents an exception thrown to indicate that a json element with the given name
+    /// was not found, even though it was expected to exist under the given parent json element.
+    /// </summary>
+    internal class JsonMemberMissingException : Exception
+    {
+        /// <summary>
+        /// Creates an instance of <see cref="JsonMemberMissingException"/>.
+        /// </summary>
+        /// <param name="owningElementName">Name of the Json element that was expected to contain
+        /// a member with the given name.</param>
+        /// <param name="memberName">Name of the missing member.</param>
+        public JsonMemberMissingException(string owningElementName, string memberName)
+            : base($"Json element \"{owningElementName}\" must have a member \"{memberName}\"")
+        {
+            OwningElementName = owningElementName;
```

Comment:
If these strings can be either null or 0 length ignore, otherwise consider validating the parameters.